### ISSUE TITLE
"failed to open user account database, code %!d(<nil>)" error when ending SSH sessions on Linux nodes

### ISSUE THREAD

**Issue by webvictim** [gravitational/teleport#20083]
Expected behavior:
SSH sessions on Linux nodes should end cleanly with the message `the connection was closed on the remote side on  11 Jan 23 10:53 AST` or similar.

Current behavior:
Pressing Ctrl+D or typing `exit<enter>` to end the session generates an error:

```
gus@artemis:~$ exit
logout
Failed to launch: failed to open user account database, code: %!d(<nil>).
ERROR: Process exited with status 255
```

This happens when using either `tsh ssh` or `ssh` (with `tsh proxy ssh`) to any Linux node in my cluster running 10.3.11. The nodes running a mix of Fedora 37 and Ubuntu 20.04.2 with a mix of reverse tunnels and direct connections.

The same issue doesn't seem to be present on a 10.3.11 auth/proxy with lower versions of nodes connected (tested 10.3.2 and 10.3.5 without issue) 

Logs (nothing obvious I could see):
```
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [PROXY:AGE] Transport request: teleport-transport. leaseID:1 target:teleport.example.com:443 cluster:teleport.example.com reversetunnel/agent.go:570
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [PROXY:AGE] Received out-of-band proxy transport request for @local-node [6c3161e1-828d-4573-90e2-0e511a452830.teleport.example.com]. cluster:teleport.example.com reversetunnel/transport.go:199
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [PROXY:AGE] Handing off connection to a local "node" service. cluster:teleport.example.com reversetunnel/transport.go:274
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      conn(192.168.64.1:55030->192.168.64.200:40662, user=gus) auth attempt fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:Tbg166Aj3q1PkCXWiUf697xNFb6tBssy6KGldb+eW4Q local:192.168.64.200:40662 remote:192.168.64.1:55030 user:gus srv/authhandlers.go:290
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      conn(192.168.64.1:55030->192.168.64.200:40662, user=gus) auth attempt with key ssh-rsa-cert-v01@openssh.com SHA256:Tbg166Aj3q1PkCXWiUf697xNFb6tBssy6KGldb+eW4Q, &ssh.Certificate{Nonce:[]uint8{0xb0, 0x3f, 0x42, 0xbb, 0x4b, 0x2e, 0xef, 0xf8, 0x54, 0x8c, 0x92, 0xea, 0xbd, 0xf3, 0xbc, 0x93, 0xd7, 0x9d, 0x3c, 0x48, 0x34, 0xb8, 0xf0, 0xb2, 0x90, 0x45, 0xa4, 0xb0, 0x6a, 0xe5, 0xbb, 0xbc}, Key:(*ssh.rsaPublicKey)(0xc0022ac7d0), Serial:0x0, CertType:0x1, KeyId:"exampleuser", ValidPrincipals:[]string{"gus", "exampleuser", "deck", "-teleport-internal-join"}, ValidAfter:0x63bec352, ValidBefore:0x63bf6c4e, Permissions:ssh.Permissions{CriticalOptions:map[string]string{}, Extensions:map[string]string{"permit-X11-forwarding":"", "permit-agent-forwarding":"", "permit-port-forwarding":"", "permit-pty":"", "teleport-roles":"{\"version\":\"v1\",\"roles\":[\"admin\"]}", "teleport-route-to-cluster":"teleport.example.com", "teleport-traits":"{\"github_teams\":[\"admins\"],\"kubernetes_groups\":null,\"kubernetes_users\":null,\"logins\":[\"exampleuser\"]}"}}, Reserved:[]uint8{}, SignatureKey:(*ssh.rsaPublicKey)(0xc0022ac810), Signature:(*ssh.Signature)(0xc0022a5b40)} fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:Tbg166Aj3q1PkCXWiUf697xNFb6tBssy6KGldb+eW4Q local:192.168.64.200:40662 remote:192.168.64.1:55030 user:gus srv/authhandlers.go:293
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Successfully authenticated fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:Tbg166Aj3q1PkCXWiUf697xNFb6tBssy6KGldb+eW4Q local:192.168.64.200:40662 remote:192.168.64.1:55030 user:gus srv/authhandlers.go:379
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Checking permissions for (exampleuser,gus) to login to node with RBAC checks. srv/authhandlers.go:544
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [RBAC]      Access to node "6c3161e1-828d-4573-90e2-0e511a452830" granted, allow rule in role "admin" matched. services/role.go:2051
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [SSH:NODE]  Incoming connection 192.168.64.1:55030 -> 192.168.64.200:40662 version: SSH-2.0-Go, certtype: "user" sshutils/server.go:484
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [KEEPALIVE] Starting keep-alive loop with interval 30s and max count 3. srv/keepalive.go:64
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request env, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU             Will create new session for SSH connection 192.168.64.1:55030. srv/ctx.go:561
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request pty-req, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Requested terminal "xterm-256color" of size {236 93} id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser srv/termhandlers.go:77
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU             Set permissions on /dev/pts/2 to 1000:5 with mode -rw-------. srv/term.go:406
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [SESSION:N] Unable to update window size, no session found in context. srv/sess.go:345
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU             Will create new session for SSH connection 192.168.64.1:55030. srv/ctx.go:561
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU [NODE]      Handling request shell, want reply true. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1549
Jan 11 10:50:59 zeus.example.com teleport[635050]: 2023-01-11T10:50:59-04:00 DEBU             Attempting GET teleport.example.com:443/webapi/find webclient/webclient.go:127
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Attempting to create session tracker session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1759
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Using async streamer for session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1265
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 INFO [NODE]      Creating (interactive) session bd14ad22-4ff4-4a99-a736-48639f4df026. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser srv/sess.go:263
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 INFO [AUDIT]     session.start addr.remote:192.168.64.1:55030 cluster_name:teleport.example.com code:T2000I ei:0 event:session.start initial_command:[] login:gus namespace:default proto:ssh server_hostname:zeus server_id:6c3161e1-828d-4573-90e2-0e511a452830 type:node session_recording:node sid:bd14ad22-4ff4-4a99-a736-48639f4df026 size:236:93 time:2023-01-11T14:51:00.86Z uid:ea0abd53-3f9e-40c7-9250-4c18b7fed0f1 user:exampleuser events/emitter.go:263
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Tracking participant: 2bd1ad14-d765-467d-a138-8c0aa87ccd25 session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1547
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 INFO [SESSION:N] New party ServerContext(192.168.64.1:55030->192.168.64.200:40662, user=gus, id=3) party(id=2bd1ad14-d765-467d-a138-8c0aa87ccd25) joined session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1568
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Launching session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:918
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Waiting for continue signal session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1033
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Got continue signal session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1038
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU             Will join session bd14ad22-4ff4-4a99-a736-48639f4df026 for SSH connection 192.168.64.1:55030. srv/ctx.go:559
Jan 11 10:51:00 zeus.example.com teleport[635050]: 2023-01-11T10:51:00-04:00 DEBU [SESSION:N] Starting poll and sync of terminal size to all parties. session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1432
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SESSION:N] Copying from PTY to writer completed with error read /dev/ptmx: input/output error. session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:964
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [SESSION:N] Stopping session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:655
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [NODE]      Exec request ("/proc/self/exe") complete: 255 id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser regular/sshserver.go:1529
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SESSION:N] Copying from reader to PTY completed with error <nil>. session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:972
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [AUDIT]     session.end addr.remote:192.168.64.1:55030 cluster_name:teleport.example.com code:T2004I ei:5 enhanced_recording:false event:session.end interactive:true login:gus namespace:default participants:[exampleuser] proto:ssh server_hostname:zeus server_id:6c3161e1-828d-4573-90e2-0e511a452830 type:node session_recording:node session_start:2023-01-11T14:50:59.887894696Z session_stop:2023-01-11T14:51:02.905498987Z sid:bd14ad22-4ff4-4a99-a736-48639f4df026 time:2023-01-11T14:51:02.906Z uid:697ae717-b19e-41cc-bb76-e8362ddd0c6c user:exampleuser events/emitter.go:263
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [NODE]      Releasing associated resources - context has been closed. id:3 local:192.168.64.200:40662 login:gus remote:192.168.64.1:55030 teleportUser:exampleuser srv/monitor.go:266
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SESSION:N] Stopping poll and sync of terminal size to all parties. session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1460
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SESSION:N] Copying from Party 2bd1ad14-d765-467d-a138-8c0aa87ccd25 to session writer completed with error <nil>. session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1577
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [TERM:LOCA] Closed PTY srv/term.go:299
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SSH:NODE]  Closed connection 192.168.64.1:55030. sshutils/server.go:489
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [SESSION:N] Closing session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:684
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [SESSION:N] Closing party 2bd1ad14-d765-467d-a138-8c0aa87ccd25 srv/sess.go:1719
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [SESSION:N] Removing party ServerContext(192.168.64.1:55030->192.168.64.200:40662, user=gus, id=3) party(id=2bd1ad14-d765-467d-a138-8c0aa87ccd25) from session session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1301
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 DEBU [SESSION:N] No longer tracking participant: 2bd1ad14-d765-467d-a138-8c0aa87ccd25 session_id:bd14ad22-4ff4-4a99-a736-48639f4df026 srv/sess.go:1309
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [AUDIT]     session.data addr.remote:192.168.64.1:55030 code:T2006I ei:2.147483646e+09 event:session.data login:gus namespace:default rx:4586 server_id:6c3161e1-828d-4573-90e2-0e511a452830 sid:bd14ad22-4ff4-4a99-a736-48639f4df026 time:2023-01-11T14:51:02.972Z tx:5442 uid:c08a00a2-4ef8-4060-8ebd-51f900c4d33e user:exampleuser events/emitter.go:263
Jan 11 10:51:02 zeus.example.com teleport[635050]: 2023-01-11T10:51:02-04:00 INFO [AUDIT]     session.leave cluster_name:teleport.example.com code:T2003I ei:6 event:session.leave login:gus namespace:default server_hostname:zeus server_id:6c3161e1-828d-4573-90e2-0e511a452830 type:node sid:bd14ad22-4ff4-4a99-a736-48639f4df026 time:2023-01-11T14:51:02.972Z uid:b4d9afc8-c7ae-4dd8-985e-c7b900a1f0f6 user:exampleuser events/emitter.go:263
Jan 11 10:51:03 zeus.example.com teleport[635050]: 2023-01-11T10:51:03-04:00 DEBU [UPLOAD]    Scanned 1 uploads, started 1, found 0 corrupted in /var/lib/teleport/log/upload/streaming/default. filesessions/fileasync.go:269
Jan 11 10:51:03 zeus.example.com teleport[635050]: 2023-01-11T10:51:03-04:00 DEBU [UPLOAD]    Session upload completed. duration:16.211182ms session-id:bd14ad22-4ff4-4a99-a736-48639f4df026 filesessions/fileasync.go:435
```

Bug details:
- Teleport version: 10.3.11

**Comment by zmb3** [comment]
Same as #17769?

**Comment by zmb3** [comment]
Released in 10.3.12
